### PR TITLE
V6.0.0 rc

### LINKED ISSUE #3403: K version bump feature list (6.0.0)
Updated up to: 

- Features:
  - https://github.com/runtimeverification/k/pull/3270
  - https://github.com/runtimeverification/k/pull/3272
  - https://github.com/runtimeverification/k/pull/3279
  - https://github.com/runtimeverification/k/pull/3288
  - https://github.com/runtimeverification/k/pull/3251
  - https://github.com/runtimeverification/k/pull/3313
  - https://github.com/runtimeverification/k/pull/3259
  - https://github.com/runtimeverification/k/pull/3324
  - https://github.com/runtimeverification/k/pull/3342
  - https://github.com/runtimeverification/k/pull/3314
  - https://github.com/runtimeverification/k/pull/3349
  - https://github.com/runtimeverification/k/pull/3391
  - https://github.com/runtimeverification/k/pull/3400
  - https://github.com/runtimeverification/k/pull/3407
  - https://github.com/runtimeverification/k/pull/3417
  - https://github.com/runtimeverification/k/pull/3390
  - https://github.com/runtimeverification/k/pull/3437
  - https://github.com/runtimeverification/k/pull/3432
  - https://github.com/runtimeverification/k/pull/3438
  - https://github.com/runtimeverification/k/pull/3382
  - https://github.com/runtimeverification/k/pull/3461
  - https://github.com/runtimeverification/k/pull/3464
  - https://github.com/runtimeverification/k/pull/3454

- Bug fixes:
  - https://github.com/runtimeverification/k/pull/3263
  - https://github.com/runtimeverification/k/pull/3289
  - https://github.com/runtimeverification/k/pull/3287
  - https://github.com/runtimeverification/k/pull/3331
  - https://github.com/runtimeverification/k/pull/3032
  - https://github.com/runtimeverification/k/pull/3341
  - https://github.com/runtimeverification/k/pull/3356
  - https://github.com/runtimeverification/k/pull/3375
  - https://github.com/runtimeverification/k/pull/3354
  - https://github.com/runtimeverification/k/pull/3412
  - https://github.com/runtimeverification/k/pull/3401
  - https://github.com/runtimeverification/k/pull/3458
  - https://github.com/runtimeverification/k/pull/3459
  - https://github.com/runtimeverification/k/pull/3467

To Check:

- [x] File `./package/version` has been bumped to `NEW_MAJOR.NEW_MINOR.0`, and `./package/version.sh sub` has been invoked.
- [x] `CHANGELOG.md` has been updated with new features.
- [x] [K active projects](https://github.com/runtimeverification/k/blob/master/web/pages/projects.md) page is up-to-date (receiving commits in the last 3 months).
- [ ] Audit of warnings in regression test-suite and tutorial (inspect CI output for unintentional warnings).
- [ ] Audit package dependencies in `package/` (look at various package/dependency lists in `package/` subdirectory and think whether any of them is now not needed).

### PR DESCRIPTION
**Pull request by radumereuta**
K v6.0.0
Bump in the major version to signify the big changes in the last couple of years.
Most notably the removal of the Java backend.
All symbolic execution will be handled by the Haskell backend and concrete execution by the llvm backend.

This needs to be correlated with the blog post.

### CODE REVIEW

**Comment by Robertorosmaninho** [comment]
@radumereuta, what is the current status of this?

**Comment by radumereuta** [comment]
Waiting on the blog post to be ready. Once that is publishable we will merge this and make the blog post public.
Ping @reekamaharaj

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,54 @@
...
+=================
+
+Features
+--------
+- Removed the Java backend. From now on, all symbolic execution will be handled by the
+  haskell backend and concrete execution by the llvm backend. Alongside that we also
+  removed the `kprove-legacy` and the `kbmc` tool. Bounded model checking capabilities
+  have been added to the pyk library.
+
+- Deprecated the `--directory` option. Use `--output-definition` instead.
```
Comment (dwightguth) [inline CHANGELOG.md]: This needs to be updated because there are now different flags for kompile and krun

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,54 @@
...
+
+- Introduced `--execute-to-branch` to `krun` when using the LLVM backend.
+
+- Created a hidden category for advanced options: `--help-hidden`. This should make it
+  easier to read the help menu for `kompile`.
+
+- Added attribute validation. From now on, you will have to use `group(_)` to tag a
+  production. This can be enabled with `--pedantic-attributes` but it will soon become
```
Comment (dwightguth) [inline CHANGELOG.md]: I believe this is now the default.

**Review by Baltoli — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,54 @@
...
+- Add `--temp-dir` option to specify where to store all the temp files created at
+  runtime. This can avoid some issues with accumulating files and readonly restrictions.
+
+- Added a new builtin type `RangeMap`, a map whose keys are stored as ranges, bounded
+  inclusively below and exclusively above. Contiguous or overlapping ranges that
+  map to the same value are merged into a single range.
+
+- Added a new, experimental backend for faster symbolic execution, called the **booster**.
+  This works by invoking the faster parts of the concrete backend and only rely on the
+  haskell backend when needed for symbolic steps.
```
Comment (Robertorosmaninho) [inline CHANGELOG.md]: This is not ready yet inside the K repository and will not be packaged on the K6, even though it exists in another repository.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,54 @@
...
+
+
+Misc/Bug Fixes
+--------------
+- Fix some issues related to unicode characters  not being parsed correctly.
```
Comment (Robertorosmaninho) [inline CHANGELOG.md]: extra space here

File: CHANGELOG.md (hunk 3 above)
Comment (radumereuta) [reply CHANGELOG.md]: Considering it's so new, better not mention it anyway. I removed it from the changelog.

**Review by Robertorosmaninho — APPROVED** [review]

**Pull request merged**